File: application/src/main/java/whenyourcar/application/converter/chat/ChatMessageConverter.java

package whenyourcar.application.converter.chat;

import org.springframework.data.domain.Page;
import org.springframework.stereotype.Component;
import whenyourcar.application.dto.chat.message.ChatMessageDto;
import whenyourcar.application.dto.chat.message.ChatMessageResponse;
import whenyourcar.domain.entity.Chat;
import whenyourcar.domain.entity.Room;
import whenyourcar.domain.entity.User;
import whenyourcar.domain.query.chat.SearchChatsQuery;

import java.util.UUID;

@Component
public class ChatMessageConverter {
    public ChatMessageDto toChatMEssageDto(String id,
                                           Long userId,
                                           Long roomId,
                                           String message) {
        return ChatMessageDto.builder()
                .id(id)
                .message(message)
                .roomId(roomId)
                .sender(userId)
                .build();
    }

    public Chat toChatDto(Room room,
                          User user,
                          String message){
        return Chat.builder()
                .id(UUID.randomUUID().toString())
                .message(message)
                .room(room)
                .user(user)
                .build();
    }

    public ChatMessageResponse.ChatMessageResponseSetDto toGetMessageDto(
            Page<SearchChatsQuery> chats,
            Long roomId,
            User sender
    ){
        return ChatMessageResponse.ChatMessageResponseSetDto.builder()
                .chatMessages(chats)
                .senderPicture(sender.getPicture())
                .senderName(sender.getName())
                .roomId(roomId)
                .build();
    }
}

File: application/src/main/kotlin/howmuchcar/application/converter/chat/ChatMessageConverter.kt
package howmuchcar.application.converter.chat

import howmuchcar.application.dto.chat.ChatMessage
import howmuchcar.application.dto.chat.ChatMessageResponse
import howmuchcar.domain.entity.Chat
import howmuchcar.domain.entity.Room
import howmuchcar.domain.entity.User
import howmuchcar.domain.query.chat.SearchChatsQuery
import howmuchcar.domain.query.user.SearchOtherUserQuery
import org.springframework.data.domain.Page
import org.springframework.stereotype.Component
import java.util.*

@Component
class ChatMessageConverter {
    fun toChatMessage(id: String, userId: Long, roomId: Long, message:String) : ChatMessage {
        return ChatMessage(
            chatId = id,
            message = message,
            senderId = userId
        )
    }

    fun toChat(room: Room, user: User, message:String): Chat {
        return Chat.builder()
            .id(UUID.randomUUID().toString())
            .message(message)
            .room(room)
            .user(user)
            .build()
    }

    fun toGetMessageDto(
        chats: Page<SearchChatsQuery>,
        roomId: Long,
        sender: SearchOtherUserQuery
    ): ChatMessageResponse {
        return ChatMessageResponse(
            chatMessages = chats,
            senderPicture = sender.picture,
            senderName = sender.name,
            roomId = roomId
        )
    }
}